### PR TITLE
Implement dstack function across all backends

### PR DESCRIPTION
**Pull request by Shi-pra-19**
Added new `dstack` ops, which stacks a sequence of tensors along the third axis (depth-wise stacking).
Supported across all backends(Numpy, Tensorflow, Torch, JAX, OpenVINO).

### CODE REVIEW

File: keras/src/backend/tensorflow/numpy.py [hunk 1]
```diff
@@ -1478,6 +1478,27 @@ def dot(x1, x2):
...
+        shape = x.shape
+        if len(shape) == 0:
+            x = tf.reshape(x, (1, 1, 1))
+        elif len(shape) == 1:
+            x = tf.expand_dims(x, axis=0)
+            x = tf.expand_dims(x, axis=2)
+        elif len(shape) == 2:
+            x = tf.expand_dims(x, axis=2)
+        xs_reshaped.append(x)
+    return tf.concat(xs_reshaped, axis=2)
```
[the remark this answers is not in the document]
Comment (Shi-pra-19) [reply keras/src/backend/tensorflow/numpy.py:1499]: done.

File: keras/src/ops/numpy.py [hunk 2]
```diff
@@ -2837,6 +2837,64 @@ def dot(x1, x2):
...
+    This is equivalent to concatenation along the third axis after 2-D tensors
+    of shape `(M, N)` have been reshaped to `(M, N, 1)` and 1-D tensors of shape
+    `(N,)` have been reshaped to `(1, N, 1)`.
+
+    Args:
+        xs: Sequence of tensors.
+
+    Returns:
+        The tensor formed by stacking the given tensors.
+    """
```
[the remark this answers is not in the document]
Comment (Shi-pra-19) [reply keras/src/ops/numpy.py:2908]: done.
